testsuite/features/support/http_client.rb

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2022 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+require 'faraday'
+
+# Wrapper class for HTTP client library (Faraday)
+class HttpClient
+  def initialize(host)
+    puts 'Activating HTTP API'
+    @http_client = Faraday.new('https://' + host)
+  end
+
+  def call(name, params)
+    # Adjust method name
+    name.sub!('auth.', '') if ['auth.login', 'auth.logout'].include?(name)
+    long_name = '/rhn/manager/api/' + name.tr('.', '/')
+    short_name = name.split('.')[-1]
+
+    # Get session cookie from previous calls
+    if params.nil?
+      session_cookie = nil
+    else
+      session_cookie = params[:sessionKey]
+      params.delete(:sessionKey)
+    end
+
+    # Call API
+    if short_name.start_with?('list', 'get', 'is', 'find') || ['logout', 'errata.applicableToChannels'].include?(name)
+      # GET
+      unless params.nil?
+        long_name += '?'
+        params.each do |key, value|
+          long_name += '&' unless long_name[-1] == '?'
+          long_name += "#{key}=#{value}"
+        end
```

Comment:
It also looks like a method to be extracted

Reply:
here it would make a bit more sense, ok